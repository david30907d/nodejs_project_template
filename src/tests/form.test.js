import React from 'react';
import FormExampleFieldErrorLabel from '../form.js';
import renderer from 'react-test-renderer';

test('Test Forum components', () => {
  const component = renderer.create(
    <FormExampleFieldErrorLabel/>
  );
  let forum = component.toJSON();
  expect(forum).toMatchSnapshot();
});