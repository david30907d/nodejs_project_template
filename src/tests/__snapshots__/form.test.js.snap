// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Forum components 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <label
      htmlFor="form-input-first-name"
    >
      Application
    </label>
    <div
      className="ui fluid input"
    >
      <input
        aria-describedby={null}
        id="form-input-first-name"
        name="Application"
        onChange={[Function]}
        placeholder="Name of camera"
        type="text"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label>
      Source
    </label>
    <div
      className="ui fluid input"
    >
      <input
        aria-describedby={null}
        name="Source"
        onChange={[Function]}
        placeholder="rtsp://ip"
        type="text"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label>
      Codec
    </label>
    <div
      className="ui fluid input"
    >
      <input
        aria-describedby={null}
        name="Codec"
        onChange={[Function]}
        placeholder="rtsp://ip"
        type="text"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label>
      Width
    </label>
    <div
      className="ui fluid input"
    >
      <input
        aria-describedby={null}
        name="Width"
        onChange={[Function]}
        placeholder="In width"
        type="text"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label>
      Height
    </label>
    <div
      className="ui fluid input"
    >
      <input
        aria-describedby={null}
        name="Height"
        onChange={[Function]}
        placeholder="In cm"
        type="text"
      />
    </div>
  </div>
  <div
    className="ui success message"
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Form Completed
      </div>
      <p>
        You've finish all the settings of BerryNet!
      </p>
    </div>
  </div>
  <div
    className="ui error message"
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Form incomplete
      </div>
      <p>
        All fields are required!
      </p>
    </div>
  </div>
  <button
    className="ui button"
    onClick={[Function]}
  >
    Submit
  </button>
</form>
`;
